### DISCUSSION TITLE
Custom Edit Form Action (one column only)

### DISCUSSION THREAD

**Discussion by Hegabovic** [filamentphp/filament#12524] [Help]
### Package

Actions

### Package Version

v3.0.0

### How can we help you?

i have this code in my table action to create a custom form to enable add/edit only one column in the current record 
```
Tables\Actions\Action::make('edit_section')
                        ->name(__('filament-actions::edit.single.label'))
                        ->modalHeading("Add Keys")
                        ->modalDescription("Add keys for specific need")
                        ->icon('heroicon-m-pencil-square')
                        ->form([
                            Repeater::make('extra_fields')
                                ->schema([
                                    TextInput::make('name')
                                        ->label(__('filament-dynamic-components-management::backend.widgets.actions.name')),
                                    TextInput::make('value')
                                        ->label(__('filament-dynamic-components-management::backend.widgets.actions.value')),
                                ])
                                ->columns(2)

                        ])->action(function (Section $section, array $data):void
                        {
                            $section->extra_fields = $data['extra_fields'];
                            $section->save();

                            Notification::make()
                                ->title("lolo")
                                ->success()
                                ->send();
                        })
```

but the thing is that it adds the element but when i reopen it , 
expected : it should bring the data saved in the column and enable editing over it 
issue : it doesn't show the data in the column and only view it as follow and never shows the data saved in the column 

![Screenshot from 2024-04-28 10-18-25](https://github.com/filamentphp/filament/assets/56083879/cdb85b5c-293a-47af-b8e8-01670a771e37)

**Comment by Hegabovic** [comment]
@leandrocfe can you provide help here ?

**Comment by micraux** [comment]
Maybe you have to [refresh](https://filamentphp.com/docs/3.x/panels/pages#refreshing-form-data) the form?

**Comment by e2d2-dev** [comment]
Hi,

you should use ->fillForm(fn($record) => $record->toArray()) on the action. Thats a blank action with to prefill like the regular EditAction


ciao ed
